app/exchanges/volume_tracking/message_builder.rb

```diff
@@ -13,35 +13,57 @@ class MessageBuilder < Message::Message
     #    "volume"=>1.5, "concentration"=>10.0, "insertSize"=>100, "aliquotType"=>"primary",
     #    "aliquotUuid"=>"", "sourceType"=>"library", "sourceBarcode"=>"TRAC-2-35805",
     #    "sampleName"=>"Sample1", "usedByBarcode"=>"TRAC-2-35806", "usedByType"=>"pool"}}
+    def publish_data
+      @publish_data ||= generate_publish_data
+    end
+
+    private
 
-    def publish_data # rubocop:disable Metrics/MethodLength
-      # Memoize the data
-      return @publish_data if defined?(@publish_data)
+    def generate_publish_data
+      data = base_data
+      populate_by_source_type(data)
+      populate_used_by_type(data)
+      data
+    end
 
-      aliquot = object
-      data = { source_type: '', source_barcode: '', sample_name: '',
-               used_by_type: 'none', used_by_barcode: '',
-               aliquot_uuid: aliquot.uuid,
-               message_uuid: SecureRandom.uuid }
+    def base_data
+      {
+        source_type: '',
+        source_barcode: '',
+        sample_name: '',
+        used_by_type: 'none',
+        used_by_barcode: '',
+        aliquot_uuid: object.uuid,
+        message_uuid: SecureRandom.uuid
+      }
+    end
 
-      case aliquot.source_type
+    def populate_by_source_type(data)
+      case object.source_type
       when 'Pacbio::Library'
         data[:source_type] = 'library'
-        data[:source_barcode] = aliquot.source.tube.barcode
-        data[:sample_name] = aliquot.source.sample_name
+        data[:source_barcode] = object.source.tube.barcode
+        data[:sample_name] = object.source.sample_name
+      when 'Pacbio::Pool'
+        data[:source_type] = 'pool'
+        data[:source_barcode] = object.source.tube.barcode
```

Comment:
Do we need to fill in sample name here? Good to ask @stevieing - with multiple samples in case of a pool, may be string which joins a friendly name for all?

Reply:
I was expected this comment, as I am sure, the object does not actually contain a sample name property @stevieing any inputs in this, please

Reply:
A pool will have multiple samples through libraries so yes we could get the names of the samples in the libraries.

Reply:
is so how are we storing that in the table, are concatenating the sample names in one string (join by , maybe )?

Reply:
> is so how are we storing that in the table, are concatenating the sample names in one string (join by , maybe )?

Yes. We can join the with a comma I guess.